scripts/locust/service.py

```diff
@@ -33,127 +57,186 @@ def trim_url(url: str):
+            # unless specified like this
+            if response.status_code == 301 or 200 <= response.status_code < 300:
+                response.success()
+        while response.status_code == 301:
+            refresh_url = trim_url(response.headers['Location'])
+            retry_after = int(response.headers['Retry-After'])
+            self._sleep(retry_after)
+            with self.client.get(refresh_url,
+                                 name='/manifest/files',
+                                 catch_response=True,
+                                 allow_redirects=False) as response:
+                if response.status_code in (301, 302):
                     response.success()
-            while response.status_code == 301:
-                refresh_url = trim_url(response.headers['Location'])
-                retry_after = int(response.headers['Retry-After'])
-                self._sleep(retry_after)
-                with self.client.get(refresh_url, catch_response=True, allow_redirects=False) as response:
-                    if response.status_code in [301, 302]:
-                        response.success()
-
-        @seq_task(3)
-        def stop(self):
-            self.interrupt()
+
+
+class IndexTaskSet(SequentialTaskSet):
+    """
+    Browse multiple pages of the samples index
+    """
+
+    organ_filter = {"organ": {"is": ["brain"]}}
+
+    @task
+    def start_page(self):
+        browse_page(self.client, 'samples', {}, size=15)
+
+    @task
+    def select_brain(self):
+        browse_page(self.client, 'samples', self.organ_filter, **browser_search_params)
+
+    @task
+    def next_pages(self):
+        url = endpoint('/index/samples',
+                       filters=json.dumps(self.organ_filter),
+                       **browser_search_params)
+        for _ in range(2):
+            with self.client.get(url, name='/index/samples') as response:
+                url = response.json()['pagination']['next']
+            if url is None:
+                break
+            else:
+                self.wait()
+
+
+class GA4GHTaskSet(SequentialTaskSet):
+    """
+    Test DRS and DOS endpoints after selecting a file
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.drs_uuid = None
+        self.dos_uuid = None
+        self.drs_access_ids = []
+
+    @task
+    def choose_file(self):
+        with self.client.get(endpoint('/index/files',
+                                      filters=json.dumps({"fileFormat": {"is": ["fastq", "fastq.gz"]}}),
+                                      size=15),
+                             name='/index/files') as response:
+            hit = random.choice(response.json()['hits'])
+            self.drs_uuid = hit['entryId']
+            self.dos_uuid = one(hit['files'])['uuid']
+
+    @task
+    def drs(self):
+        drs_path = '/ga4gh/drs/v1/objects/{file_uuid}'
```

Comment:
I don't see `catalog` anywhere in this script which I take to mean that the default `dcp2` catalog is used. That's fine but our DRS implementation only works with with files in the `dcp1` catalog. The files UUID happen to match between both catalog which is probably why this doesn't fail but we shouldn't depend on this incidental agreement between the catalogs.

I think the script should be parameterized with the `catalog` (None should be supported, too, to test the default catalog) and our DRS/DOS endpoints should be exercised on the `dcp1` catalog.